packages/polling-controller/src/StaticIntervalPollingController.ts

```diff
@@ -48,27 +48,25 @@ function StaticIntervalPollingControllerMixin<TBase extends Constructor>(
 
       const key = getKey(networkClientId, options);
       const existingInterval = this.#intervalIds[key];
-      this._stopPollingByPollingTokenSetId(key);
 
-      this.#intervalIds[key] = setTimeout(
+      // eslint-disable-next-line no-multi-assign
+      const intervalId = (this.#intervalIds[key] = setTimeout(
         async () => {
           try {
             await this._executePoll(networkClientId, options);
           } catch (error) {
             console.error(error);
           }
-          this._startPollingByNetworkClientId(networkClientId, options);
+          if (intervalId === this.#intervalIds[key]) {
+            this._startPollingByNetworkClientId(networkClientId, options);
+          }
         },
         existingInterval ? this.#intervalLength : 0,
-      );
+      ));
     }
 
     _stopPollingByPollingTokenSetId(key: PollingTokenSetId) {
-      const intervalId = this.#intervalIds[key];
-      if (intervalId) {
-        clearTimeout(intervalId);
-        delete this.#intervalIds[key];
-      }
+      delete this.#intervalIds[key];
```

Comment:
Do we not still need to clear these timeouts? I realize that it won't really stop the loop since each loop gets a new setTimeout, but I think this may result in a memory leak?

Reply:
Actually nvm it seems all of the references expire after the timeout expires

Reply:
either we need to clear the interval OR we need to add an additional guard before _executePoll

I think it makes most sense to restore this clearTimeout. We may want to restore the stop inside the start as well